pkg/ebpf/tracee.go

```diff
@@ -240,6 +240,30 @@ func New(cfg Config) (*Tracee, error) {
 		}
 	}
 
+	// Clean up signatures not chosen for tracing
+	// Looping from the back to avoid bounds checking
+	for i := len(t.config.EngineConfig.Signatures) - 1; i > 0; i-- {
+		sig := t.config.EngineConfig.Signatures[i]
```

Comment:
You can use range instead, build the signatures list and assign to `t.config.EngineConfig.Signatures` at the end.